### PR TITLE
STORE-1160 Adding time in lifecycle at publisher

### PR DESCRIPTION
**Pull request by lahirue**
Show Time in lifecycle in publisher UIs. And color box will be notified bellow the asset icon based on lifecycle xml checkpoints. When hovering the color box, current state duration time will be visible. User can enable or disable this feature.

### CODE REVIEW

File: apps/publisher/modules/lifecycle/lifecycle-api.js [hunk 1]
```diff
@@ -409,6 +409,13 @@ var error = '';
         state.isLCActionsPermitted = isLCActionsPermitted(asset, options, req, res, session);
         state.checkItems = setCurrentCheckItemState(state.checkItems, lcCheckedStates, state.isLCActionsPermitted);
         state.approvedActions = setAvailableApprovedActions(state.approvedActions, lcCheckedStates);
+        var isStateDurationEnable = rxtManager.isLCStateDurationStateEnabled(options.type);
```
Comment (splinter) [inline apps/publisher/modules/lifecycle/lifecycle-api.js]: Eagerly set state.isLCStateDurationEnabled = false;

Then we can get rid of the else statement

File: apps/publisher/themes/default/partials/lifecycle-state-duration.hbs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<!--
```
Comment (splinter) [inline apps/publisher/themes/default/partials/lifecycle-state-duration.hbs]: Shall we have this as Handlebars comments, other wise these will appear in page.

File: jaggery-modules/rxt/module/scripts/asset/asset.js [hunk 3]
```diff
@@ -1633,6 +1633,38 @@ var asset = {};
...
+    /**
+     * This method is used to format a timestamp to 'd:h:m:s'.
+     *
+     * @param duration  timestamp duration in milliseconds.
+     * @return String   formatted time duration to 'd:h:m:s'.
+     */
+    AssetManager.prototype.formatLCStateTimeDuration = function (duration) {
+        var mil = duration;
+        var timeDuration;
+        var seconds = (mil / 1000) | 0;
```
Comment (splinter) [inline jaggery-modules/rxt/module/scripts/asset/asset.js]: Shall we have these magic numbers as constants (e.g. SECONDS_IN_MINUTE = 60 )

File: jaggery-modules/rxt/module/scripts/asset/asset.js [hunk 4]
```diff
@@ -1650,15 +1682,27 @@ var asset = {};
                     var secondChar = str[1].charAt(0).toLowerCase();
                 } catch (err) {
                     if (typeof secondChar === 'undefined') {
-                        if (str.length > 2){
-                            secondChar =str[0].charAt(1);
+                        if (str.length > 2) {
```
Comment (splinter) [inline jaggery-modules/rxt/module/scripts/asset/asset.js]: White space characters

File: jaggery-modules/rxt/module/scripts/asset/asset.js [hunk 5]
```diff
@@ -1667,6 +1711,49 @@ var asset = {};
         return asset;
     };
 
+    /***
+     * This method is used get the Current Lifecycle State Duration Meta Data.
+     * @param asset
+     * @returns {{}} json object that contains required metadata
+     */
+    AssetManager.prototype.getLCStateDurationDetails = function (asset) {
+        var details = {};
```
Comment (splinter) [inline jaggery-modules/rxt/module/scripts/asset/asset.js]: Shall we refactor the method as follows:

var details = {} ;
details.lifecycleStateDuration = "time not available";

if ( !asset.id ) {
 log.error('Some error message here');
  return details;
}

if( !asset.lifecycle ) {
   log.error('Asset does not have a lifecycle ! ');
  return details ;
}

try {

} catch (e) {

}

return details;

File: apps/publisher/modules/lifecycle/lifecycle-api.js (hunk 1 above)
Comment (lahirue) [inline apps/publisher/modules/lifecycle/lifecycle-api.js]: Thank you Sameera,
This entire code block has been refactored with new time in lifecycle feature.

File: apps/publisher/themes/default/partials/lifecycle-state-duration.hbs (hunk 2 above)
Comment (lahirue) [inline apps/publisher/themes/default/partials/lifecycle-state-duration.hbs]: Thank you Sameera,
This entire code block has been refactored with new time in lifecycle feature.

File: jaggery-modules/rxt/module/scripts/asset/asset.js (hunk 3 above)
Comment (lahirue) [inline jaggery-modules/rxt/module/scripts/asset/asset.js]: Thank you Sameera,
This entire code block has been refactored with new time in lifecycle feature.

File: jaggery-modules/rxt/module/scripts/asset/asset.js (hunk 4 above)
Comment (lahirue) [inline jaggery-modules/rxt/module/scripts/asset/asset.js]: Thank you Sameera,
This entire code block has been refactored with new time in lifecycle feature.

File: jaggery-modules/rxt/module/scripts/asset/asset.js (hunk 5 above)
Comment (lahirue) [inline jaggery-modules/rxt/module/scripts/asset/asset.js]: Thank you Sameera,
This entire code block has been refactored with new time in lifecycle feature.

**Comment by daneshk** [comment]
Closing the PR, since changes are conflicting with the latest changes and according to the review comments this fixes are outdated. Please fix the conflicts and reopen the PR, if you need to add this change

**Pull request closed without merging**